### ISSUE TITLE
Enforce handshake message ordering

### ISSUE THREAD

**Issue by bifurcation** [bifurcation/mint#25]
Handshake messages are required to arrive in a particular order (with some messages optional): ClientHello, ServerHello, EncryptedExtensions?, etc.  Right now, we will read the servers messages in any order.  We should enforce that they come in the proper order.

**Comment by ekr** [comment]
@bifurcation I don't think this is true. The state machine should enforce correct ordering at this point.
